R/utils.R

```diff
@@ -0,0 +1,325 @@
+# Helper functions for do_read_bib()
+
+#' Parse (at)string keys
+#'
+#' Returns a table with all the string values and their value to be used on the
+#' `bib` file.
+#'
+#' @noRd
+parse_strings <- function(map_string,
+                          stringlines) {
+
+  # This needs to be done in a loop, since strings can depend also
+  # on values of previous strings
+
+  parsed_string <- NULL
+
+  for (i in seq_len(length(stringlines))) {
+    row <- map_string[i, ]
+    singleline <- stringlines[i]
+
+    map_braces <- check_balanced_braces(singleline, row$init)
+
+    # Extract string between first and last brace
+
+    f <- map_braces$pos[1]
```

Comment:
Could we improve the variable names? Maybe:

`f = first_bracket_position`
`l = last_bracket_position`.

Reply:
Done